frontend/src/components/AppNavBar.vue

```diff
@@ -13,7 +13,7 @@
                 <router-link tag="h3" v-if="!isFirebaseAuthorised" class="link-item" to="/login">Login</router-link>
                 <router-link tag="h3" v-if="!isFirebaseAuthorised" class="link-item" to="/signup">Sign Up</router-link>
                 <router-link tag="h3" v-if="isFirebaseAuthorised && !hasProfile" class="link-item" to="/create-profile">Create Profile</router-link>
-                <router-link tag="h3" v-if="isLoggedIn" class="link-item" to="/profile">Profile</router-link>
+                <router-link tag="h3" v-if="isLoggedIn && hasProfile" class="link-item" to="/profile">Profile</router-link>
```

Comment:
This is a redundant check because `isLoggedIn = hasProfile && isFirebaseAuthorised`